case-lib/lib.sh

```diff
@@ -40,10 +40,26 @@ if [ ! "$SOFCARD" ]; then
 		awk '/sof-[a-z]/ && $1 ~ /^[0-9]+$/ { $1=$1; print $1; exit 0;}')
 fi
 
-func_lib_setup_kernel_checkpoint()
+setup_kernel_check_point()
 {
-    # shellcheck disable=SC2034 # external script will use it
-    KERNEL_CHECKPOINT=$(date +%s)
+    local check_point
+    check_point="$1"
+    case "$check_point" in
+        "boot")
+            # shellcheck disable=SC2034 # external script will use it
+            KERNEL_CHECKPOINT="boot"
+            ;;
+        "now")
+            # There is a 5s delay between two cases by sof-framework,
+            # make the check point 6s earlier to avoid log loss.
```

Comment:
This is too brittle. The simpler and safer way to solve this problem is to set checkpoints only at (almost) the same time as when collecting logs and never at any other time.

More importantly: I think this is an important but different problem, not directly related to this Pull Request, so best left aside for now.

Reply:
@marc-hb I was not able to understand your feedback. 

a) please describe when checkpoints should be set.

b) why can it be done in a separate PR, since it's a key functionality to enable journalctl.

Reply:
@marc-hb Before implementing this, I was thinking to use a new Global Variable to collect boot logs, too. But then I found that this can be integrated in the same function, `setup_kernel_check_point`,  to provide a unified API for test case. 

A test case designer can decide whether they want to collect logs from boot, or from now. When you say another check point, I think you mean `setup_kernel_check_point "boot"`, Actually this is only used in the first test case currently.  `Moving forward` principle is not influenced, because we give test case the flexibility setup check point. For this reason, we take "boot" as a virtual check point.

With current code, the command for collecting log from boot is: `journalctl -k -q --no-pager --utc --output=short-monotonic 
--no-hostname --boot `,  the command for collecting log from check point is: `journalctl -k -q --no-pager --utc --output=short-monotonic --no-hostname --boot  --since=@CHECK_POINT`, already good enough to handle both conditions.

By the way, the `--boot` option means collect log of the latest boot, it doesn't mean collect boot logs.

@plbossart @marc-hb **On the potential log loss.**

sof-framework will add a 5s delay between two test cases, we are not sure there will be new logs in this 5s period, so I make the checkpoint 6s earlier with KERNEL_CHECKPOINT=$(($(date +%s) - 6)).  Actually, no case is running on DUT in this 5s period, very low possibility to have kernel logs, I think. The point here becomes, should we care and collect very likely unrelated and sometime may be empty logs between test cases? With 6 here, we may have 1s log duplicated. We can choose 0, 1, 2, 3, 4, 5 or 6  to decrease log loss. In my opinion now, 5 may be the best with very low possibility of log loss and no log duplication.

Reply:
>  in the same function, setup_kernel_check_point, to provide a unified API for test case.

I don't think it's good to have a unified API for two different things, especially not when the API is named after only one of the two things. 

> Actually this is only used in the first test case currently.

So it's not great either to change and require almost every invocation to over-specify "now" and introduce the abstract concept of a "virtual" checkpoints that are not checkpoints, all this for just one special case. BTW it should be possible to run tests in any order.

In fact the only test that uses `setup_kernel_check_point "boot"` in this PR is apparently `verify-firmware-presence.sh` That test is super short and simple and merely making sure that `/lib/firmware/intel/sof/platform.ri` exists, so it should not even check kernel logs at all, kernel logs are not relevant for such a test. Is it really the only test that needed `kern.log`?

In the future we probably want a test that does nothing and checks boot logs but that would be a different test, it shouldn't be confused with checking `/lib/firmware`.

> By the way, the --boot option means collect log of the latest boot, it doesn't mean collect boot logs.

Sorry I used both somewhat interchangeably.

Reply:
> The point here becomes, should we care and collect very likely unrelated and sometime may be empty logs between test cases?

Good question but IMHO not new and independent from this PR. One problem at a time.